sktime/transformations/series/difference.py

```diff
@@ -213,8 +254,19 @@ def _transform(self, X, y=None):
             transformed version of X
         """
         Xt = _diff_transform(X, self._lags)
-        if self.drop_na:
+
+        na_handling = self._na_handling
+        if na_handling == "drop_na":
             Xt = Xt.iloc[self._cumulative_lags[-1] :]
+        elif na_handling == "fill_zero":
+            Xt.iloc[: self._cumulative_lags[-1]] = 0
+        elif na_handling == "keep_na":
+            pass
+        else:
+            raise RuntimeError(
```

Comment:
hm is that really a `RuntimeError`? Maybe just `AttributeError`?

Reply:
well, it's an unexpected condition, just a very defensive error.
If it's reached, it's unclear what it would be imo.

Reply:
This is a ValueError not RuntimeError. From the docs a value error is:
'Raised when an operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as IndexError'.

Reply:
I agree with @chrisholder

Reply:
ok, changing to ValueError